Rework episode 02 - cut down on Git content

anenadic opened the pull request:
This PR significantly cuts down on content from the previous version of the Git episode. I did not include the virtual envs in this episode and have kept it separate as this episode still feels too big. Wonder it we need to cut it further or it is OK as is?

Screenshots are not clear - need redoing.

tobyhodges commented on episodes/02-version-control.md:
```diff
@@ -36,41 +37,85 @@ https://github.com/carpentries-incubator/astronaut-data-analysis-not-so-fair/tre
 Git is used mostly for managing code when developing software, but it can track *any* files --
 and is particularly effective with text-based files (e.g. source code like `.py`, `.c`, `.r`, but also `.csv`, `.tex` and more).
 
+Git allows multiple people to work on the same project (even the same file) at the same time.
```
```suggestion
Git helps multiple people work on the same project (even the same file) at the same time.
```

tobyhodges commented on episodes/02-version-control.md:
```diff
@@ -36,41 +37,85 @@ https://github.com/carpentries-incubator/astronaut-data-analysis-not-so-fair/tre
+
+:::::: callout 
+
+### What is a version control system?
```
```suggestion
### Version control refresher

#### What is a version control system?
```

tobyhodges commented on episodes/02-version-control.md:
```diff
@@ -36,41 +37,85 @@ https://github.com/carpentries-incubator/astronaut-data-analysis-not-so-fair/tre
 but for *any* files you want, and a lot more powerful and flexible.
 
-### Why use a version control system?
+**Why use a version control system?**
```
```suggestion
#### Why use a version control system?
```

tobyhodges commented on episodes/02-version-control.md:
```diff
@@ -100,17 +145,10 @@ We'll cover these commands in more detail -- this is just a high-level overview.
-
+::::::
 
 ### Create a new repository
```
```suggestion
### Start tracking changes with Git
```

tobyhodges commented on episodes/02-version-control.md:
````diff
@@ -387,26 +363,8 @@ $ git mv my\ code\ v2.py my_code_v2.py
- 1 file changed, 0 insertions(+), 0 deletions(-)
- rename my code v2.py => my_code_v2.py (100%)
-```
+`git mv <old> <new>` command is the equivalent of running `mv <old> <new>` immediately followed by `git add <new>`,
+so the changes have been staged automatically. We just need to commit them.
````
I think there's another advantage here: with `mv` only, Git thinks one file has been deleted and a new one created. Whereas, when you use `git mv`, Git "understands" that it's the same file but with a new name.

tobyhodges replied:
```suggestion
`git mv <old> <new>` command is the equivalent of running `mv <old> <new>` immediately followed by `git add <old> <new>`,
so the changes have been staged automatically. We just need to commit them.
```

A small change that gets the same point across that I was trying to make.

tobyhodges commented on episodes/02-version-control.md:
```diff
@@ -201,20 +193,20 @@ Untracked files:
+Git to track and store (GBs ot TBs of space telescope data) or require sensitive information we cannot share
 (for example, medical records).
 
+### Add files into repository
+
 We can tell Git to track a file using `git add`:
```
Should we demonstrate how to do these version control steps with the VS Code graphical interface? Or do you prefer to keep people working in the terminal as much as possible?

anenadic replied:
We have a separate issue for this now.

tobyhodges reviewed (commented):
Thanks @anenadic. I think you've made a good judgement of which content to cut, which to put into callouts, and which to keep. It is a good balance between refreshing learners' memories and moving quickly through to new things. I also like the focus on setting things up for an existing project.

I have left a few questions and suggestions. 

In particular, I wonder about whether we should lean more into the use of the VS Code interface. The IDE has great integration with Git/GitHub and I would be in favour of making more use of that. But you may have good reasons to want to keep learners working in the terminal. Curious to hear your thoughts.

anenadic commented:
OK, I think I have addressed all the points apart from 2, for which we have separate issues:

- redoing GitHub screenshots #205 
- instructor note on using Git from VS Code

If you agree with that what the episode looks now - we can merge and move on onto other related issues for this episode @tobyhodges ?

tobyhodges reviewed (approved):
Thanks @anenadic This is ready to merge from my perspective.
